### PR TITLE
improve instructions in templates & publish base docker images to dockerhub

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Move us closer to the vision in #2360 by describing how all non-standard-testing steps can be done without touching gradle in python modules. A following PR will describe will edit current READMEs -- this PR just edits templates to make things easier to review.  

**important note: base images will now be published to dockerhub so someone can inherit them without having to build the docker image themseslves. If this sounds good, I will start publishing the images.**

We'll still need to change READMEs one more time once we have standard tests that can be run without gradle.

## Post-merge checklist
- [ ] document criteria for publishing base images and what should be done with inheriting connectors #2457
- [ ] should we use `requirements.txt` or `setup.py`

### CODE REVIEW

File: airbyte-integrations/connector-templates/source-python/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM airbyte/integration-base-python:dev
+FROM airbyte/integration-base-python:0.1.0
```
Comment (michel-tricot) [inline airbyte-integrations/connector-templates/source-python/Dockerfile:1]: I wonder if we should `s/integration/connector` at the same time.

File: airbyte-integrations/connector-templates/source-python/setup.py.hbs [hunk 2]
```diff
@@ -30,15 +30,6 @@ setup(
...
-    setup_requires=["pytest-runner"],
-    tests_require=["pytest"],
-    extras_require={
-        # Dependencies required by the main package but not integration tests should go in main. Deps required by
-        # integration tests but not the main package go in tests. Deps required by both should go in
-        # install_requires.
-        "main": [],
-        "tests": ["airbyte-python-test", "pytest"],
-    },
+    install_requires=["airbyte-protocol", "base-python", "pytest==6.1.2"],
```
Comment (michel-tricot) [inline airbyte-integrations/connector-templates/source-python/setup.py.hbs:33]: Why don't we put everything in `requirements.txt`?

**Review by michel-tricot — COMMENTED** [review]
This is great. 

I think you should document what are the criterias for publishing a new base image, and if we do what is the process after that for the other connectors

**Review by cgardens — APPROVED** [review]

File: airbyte-integrations/connector-templates/source-generic/README.md.hbs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+Then run any of the connector commands as follows:
+```
+docker run --rm airbyte/source-{{dashCase name}}:dev spec
+docker run --rm -v $(pwd)/secrets:/secrets airbyte/source-{{dashCase name}}:dev check --config /secrets/config.json
+docker run --rm -v $(pwd)/secrets:/secrets airbyte/source-{{dashCase name}}:dev discover --config /secrets/config.json
+docker run --rm -v $(pwd)/secrets:/secrets -v $(pwd)/sample_files:/sample_files airbyte/source-{{dashCase name}}:dev read --config /secrets/config.json --catalog /sample_files/configured_catalog.json
+```
+
+#### Create credentials
+**If you are a community contributor**, follow the instructions in the [documentation](https://docs.airbyte.io/integrations/sources/{{dashCase name}})
```
Comment (davinchia) [inline airbyte-integrations/connector-templates/source-generic/README.md.hbs:40]: These instructions are great. I was wondering about this!

File: airbyte-integrations/connector-templates/source-python/setup.py.hbs (hunk 2 above)
Comment (davinchia) [reply airbyte-integrations/connector-templates/source-python/setup.py.hbs:33]: I'm also curious how we decide to put something in `setup.py` vs `requirements.txt.

**Review by davinchia — APPROVED** [review]

File: airbyte-integrations/connector-templates/source-python/setup.py.hbs (hunk 2 above)
Comment (sherifnada) [reply airbyte-integrations/connector-templates/source-python/setup.py.hbs:33]: Honestly, there's not a huge reason to use `setup.py` today. It's just a little bit more flexible when packaging data dependencies like JSON files, which we do with schema files or `spec.json`. Will follow up in a separate PR.

File: airbyte-integrations/connector-templates/source-python/Dockerfile (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/connector-templates/source-python/Dockerfile:1]: Good idea, will do in the downstream PR

**Pull request merged**